pkg/invoice/invoice.go

```diff
@@ -133,10 +151,20 @@ func invoiceForTenant(ctx context.Context, tx *sqlx.Tx, tenant db.Tenant, year i
 	}, nil
 }
 
+// rawItem is a line item with additional internal fields for querying.
+// This way we do not needlessly expose (and test) internal IDs.
+type rawItem struct {
+	Item
+	// QueryID is the id of the query that generated this item
+	QueryID string `db:"query_id"`
+	// ParentQueryID is the id of the parent-query of the query that generated this item
+	ParentQueryID sql.NullString `db:"parent_query_id"`
+}
+
 func itemsForCategory(ctx context.Context, tx *sqlx.Tx, tenant db.Tenant, category db.Category, year int, month time.Month) ([]Item, error) {
-	var items []Item
+	var items []rawItem
 	err := sqlx.SelectContext(ctx, tx, &items,
-		`SELECT queries.description,
+		`SELECT queries.description, queries.id as query_id, queries.parent_id as parent_query_id,
 				SUM(facts.quantity) as quantity, MIN(facts.quantity) as quantitymin, AVG(facts.quantity) as quantityavg, MAX(facts.quantity) as quantitymax,
 				products.unit, products.amount AS pricePerUnit, discounts.discount,
```

Comment:
```suggestion
				queries.unit, products.amount AS pricePerUnit, discounts.discount,
```
The current implementation always returns the `unit` of the product and not of the query. Is this correct? According to the documented data model products actually shouldn't have a unit..

Reply:
There is a unit on the query which is supposed to be used. I have no idea, where the unit on the product comes from. We probably should get rid of `product.uni` and ensure `query.unit` gets used instead.

Reply:
I'm not removing the `product.unit` from the database for now, as this is a breaking change.